File: 프로그래머스/Lv.0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.Arrays;

class Solution {
    public int solution(int[] array) {
        int answer = 0;

        int len = array.length;
        int[] mode = new int[1002];

        for (int j : array) {

            mode[j]++;
        }

        int max = 0;
        int maxCount = 0;

        int maxIndex = 0;
        for (int i = 0; i < mode.length; i++) {

            if (max < mode[i]) {
                max = mode[i];
                maxCount = 0;
                maxIndex = i;
            }

            if (max == mode[i]) {
                maxCount++;
            }
        }

        if (maxCount > 1) {
            return -1;
        } else {
            return maxIndex;
        }
    }
}

File: JavaScript/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    var answer = 0;

    let maxVal = Math.max(...array); 2

    //maxVal + 1개의 index를 가진 배열을 선언후 fill로 모든 인덱스를 0으로 채움
    const countArr = new Array(maxVal + 1).fill(0);

    for (let i = 0; i < array.length; i++) {
        const ind = array[i];

        countArr[ind] ++;
    }

    

    let maxCount = 0;
    maxVal = 0;
    for (const i of countArr) {

        if (i === maxVal) {
            maxCount++;
            continue
        }

        if (i > maxVal) {
            maxVal = i;
            maxCount = 0;
        }
    }

    return maxCount > 0 ? -1 : countArr.indexOf(maxVal);
}